src/main/java/demohometask/demohometask/Match.java

```diff
@@ -0,0 +1,92 @@
+package demohometask.demohometask;
+
+import org.springframework.stereotype.Component;
+
+import java.io.FileOutputStream;
+import java.io.ObjectOutputStream;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Random;
+
+@Component
+public class Match {
+    private static int round;
+    private static int numberOfRounds = 3;
+    private static String score;
+
+
+    public static List<Team> createListOfTeams(){
+        String[] names = {"John", "Kevin", "Bruno", "Max", "Mike", "Jack", "Baylor", "Villanova"};
+        String[] teams = {"Team_name_1", "Team_name_2", "Team_name_3", "Team_name_4", "Team_name_5", "Team_name_6", "Team_name_7", "Team_name_8"};
+        List<Team> listOfTeams = new LinkedList();
+
+        for(int i =0; i<names.length; i++){
+            listOfTeams.add(new Team(teams[i], names[i], names[i]));
+        }
+        return listOfTeams;
+    }
+
+    public static void start(){
+        List<Team> listOfTeams = createListOfTeams();
+        printListOfTeams(listOfTeams);
+        System.out.println("\nLet's start playing :D");
+        printResult(listOfTeams);
+    }
+
+    public static void writeObject(Team object){
+        try(ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream("D:\\IdeaProjects\\InventorSoftProjects\\demo-home-task\\src\\main\\java\\demohometask\\demohometask\\team.txt")))
+        {
+            oos.writeObject(object);
+        }
+        catch(Exception ex){
+            System.out.println(ex.getMessage());
+        }
+    }
+
+    public static void printListOfTeams(List<Team> listOfTeams){
+        System.out.println("^^^^^^^ LIST of teams ^^^^^^^");
+        for(Team team : listOfTeams){
+            System.out.println(team.getName() + ", " + team.getCapitan() + ", " + team.getCoach());
+        }
+    }
+
+    public static List<Team> play(List<Team> listOfTeams){
```

Comment:
This method contains too much logic from different places: there is UI part where you print to console, there is writeObject() that integrates with storage and there is score simulation. Please, consider spliting it into different components